src/sentry/static/sentry/app/views/organizationDiscover/sidebar/queryRead.jsx

```diff
@@ -0,0 +1,93 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Flex} from 'grid-emotion';
+
+import {t} from 'app/locale';
+import Button from 'app/components/button';
+import * as conditionsUtils from '../conditions/utils';
+import * as aggregationsUtils from '../aggregations/utils';
+
+import {Fieldset, PlaceholderText, SidebarLabel, ButtonSpinner} from '../styles';
+
+export default class QueryRead extends React.Component {
+  static propTypes = {
+    queryBuilder: PropTypes.object.isRequired,
+    isFetchingQuery: PropTypes.bool.isRequired,
+    runQuery: PropTypes.func.isRequired,
```

Comment:
nit: we should name callback functions like this with an `on`-prefix

Reply:
Updating